src/packages/cli/src/detach.ts

```diff
@@ -0,0 +1,261 @@
+
+export type DetachedInstance = {
+  instanceName: string;
+  pid: number;
+  startTime: number;
+  host: string;
+  port: number;
+  flavor: FlavorName;
+  cmd: string;
+  version: string;
+};
+
+type CodedError = Error & { code: string };
+
+const READY_MESSAGE = "ready";
+const START_ERROR = "An error ocurred spawning a detached instance of Ganache:";
+const dataPath = envPaths(`Ganache/instances`).data;
+
+async function getInstanceFilePath(instanceName: string): Promise<string> {
+  return path.join(await dataPath, instanceName);
+}
+
+/**
+ * Notify that the detached instance has started and is ready to receive requests.
+ */
+export function notifyDetachedInstanceReady() {
+  // in "detach" mode, the parent will wait until the "ready" message is
+  // received before disconnecting from the child process.
+  process.send(READY_MESSAGE);
+}
+
+/**
+ * Attempt to find and remove the instance file for a detached instance.
+ * @param  {string} instanceName the name of the instance to be removed
+ * @returns boolean indicating whether the instance file was cleaned up successfully
+ */
+export async function removeDetachedInstanceFile(
+  instanceName: string
+): Promise<boolean> {
+  const instanceFilename = await getInstanceFilePath(instanceName);
+  try {
+    await rm(instanceFilename);
+    return true;
+  } catch (err) {}
+  return false;
+}
+
+/**
+ * Attempts to stop a detached instance with the specified instance name by
+ * sending a SIGTERM signal. Returns a boolean indicating whether the process
+ * was found. If the PID is identified, but the process is not found, any
+ * corresponding instance file will be removed.
+ *
+ * Note: This does not guarantee that the instance actually stops.
+ * @param  {string} instanceName
+ * @returns boolean indicating whether the instance was found.
+ */
+export async function stopDetachedInstance(
+  instanceName: string
+): Promise<boolean> {
+  try {
+    // getDetachedInstanceByName() throws if the instance file is not found or
+    // cannot be parsed
+    const instance = await getDetachedInstanceByName(instanceName);
+
+    // process.kill() throws if the process was not found (or was a group
+    // process in Windows)
+    process.kill(instance.pid, "SIGTERM");
+  } catch (err) {
+    return false;
+  } finally {
+    await removeDetachedInstanceFile(instanceName);
+  }
+  return true;
+}
+
+/**
+ * Start an instance of Ganache in detached mode.
+ * @param  {string[]} argv arguments to be passed to the new instance.
+ * @returns {Promise<DetachedInstance>} resolves to the DetachedInstance once it
+ * is started and ready to receive requests.
+ */
+export async function startDetachedInstance(
+  argv: string[],
+  instanceInfo: {
+    flavor?: FlavorName;
+    server: { host: string; port: number };
+  },
+  version: string
+): Promise<DetachedInstance> {
+  const [bin, module, ...args] = argv;
+  const childArgs = args.filter(
+    arg => arg !== "--detach" && arg !== "--😈" && arg !== "-D"
```

Comment:
This fails with variants of `ganache --detach=1`

Reply:
🙀
It's a shame that it isn't treated as a flag, because stripping out the detach argument becomes more complicated. I've implemented it with a regex check, and covered it in tests.